File: SoftInvPersistance/src/main/java/pe/edu/pucp/softinv/dao/CitaDAO.java
package pe.edu.pucp.softinv.dao;

import pe.edu.pucp.softinv.model.Cita.CitaDTO;

import java.util.ArrayList;
import java.util.Date;

public interface CitaDAO {
    public Integer insertar(CitaDTO cita);
    public CitaDTO obtenerPorId(CitaDTO idCita);
    public Integer modificar(CitaDTO cita);
    public Integer eliminar(CitaDTO cita);
    ArrayList<CitaDTO> listarCitasPorPeriodo(Date  inicio, Date fin);
}

File: C#/SoftInvPersistance/DAO/CitaDAO.cs
using softinv.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftInv.DAO
{
    public interface CitaDAO
    {
        int insertar(CitaDTO cita);
        CitaDTO obtenerPorId(CitaDTO idCita);
        int modificar(CitaDTO cita);
        int eliminar(CitaDTO cita);
        IList<CitaDTO> listarCitasPorPeriodo(DateTime inicio, DateTime fin);
    }
}
